cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseMetricsTable.java

```diff
@@ -260,13 +337,26 @@ private ScanBuilder configureRangeScan(ScanBuilder scan, @Nullable byte[] startR
     if (filter != null) {
       List<Pair<byte[], byte[]>> fuzzyPairs = Lists.newArrayListWithExpectedSize(filter.getFuzzyKeysData().size());
       for (ImmutablePair<byte[], byte[]> pair : filter.getFuzzyKeysData()) {
-        fuzzyPairs.add(Pair.newPair(pair.getFirst(), pair.getSecond()));
+        if (rowKeyDistributor != null) {
+          saltFilterPairs(fuzzyPairs, pair);
+        } else {
+          fuzzyPairs.add(Pair.newPair(pair.getFirst(), pair.getSecond()));
+        }
       }
       scan.setFilter(new org.apache.hadoop.hbase.filter.FuzzyRowFilter(fuzzyPairs));
     }
     return scan;
   }
 
+  private void saltFilterPairs(List<Pair<byte[], byte[]>> fuzzyPairs, ImmutablePair<byte[], byte[]> pair) {
+    byte[][] firstAllDistKeys = rowKeyDistributor.getAllDistributedKeys(pair.getFirst());
```

Comment:
this should be a method of the rowkey distributor. As it relies on its internals.